src/main/com/yetanalytics/lrs/pedestal/interceptor/xapi/statements.cljc

```diff
@@ -299,6 +282,8 @@
              s-count 0]
         (if-let [x (a/<! statement-result-chan)]
           (case x
+            ;; terminate on error producing invalid JSON
+            ::lrsp/async-error nil
```

Comment:
So where exactly is `::lrsp/async-error` returned from?

Reply:
_Wouldn't you like to know..._

Async impls stream their responses over a core.async channel. Previously, the impl was expected to only return data and close the channel when done. If an error was encountered mid-stream (like a failed s3 get) the only thing they could do was not send the data and close the channel, which resulted in responses that were structurally valid but did not contain the data requested.

Now, by passing `::lrsp/async-error` an async impl can signal to the body rendering functions that an error has occurred, and that they should immediately stop, resulting in a malformed response. Once you're streaming the body you can't change the request status, so it is preferable to send a 200 response with a body that won't parse, rather than one that is structurally OK but is missing data.

see https://github.com/yetanalytics/third/blob/LRS-56/src/main/cloud_lrs/impl.cljc#L189-L204 for an (in-progress) example.

Reply:
So to answer my question, this keyword is returned from somewhere _in the application code_.

Also a separate question, but it's okay to just return a keyword with no error metadata, right? Since you can't return any extra error data in your HTTP response since it's still technically a 200 OK.

Reply:
> ...it's okay to just return a keyword with no error metadata, right?

Yeah I thought about this, there just isn't anything useful the body renderers can do with any more info, so I felt an opaque keyword was appropriate.